src/state_manager/mod.rs

```diff
@@ -1582,7 +1627,8 @@ pub fn apply_block_messages<DB>(
     tipset: Arc<Tipset>,
     mut callback: Option<impl FnMut(MessageCallbackCtx<'_>) -> anyhow::Result<()>>,
     enable_tracing: VMTrace,
-) -> Result<CidPair, anyhow::Error>
+    store_events: bool,
```

Comment:
I'd recommend not using a boolean here - the method has already plenty of parameters and it's cumbersome for the reader what is `foo(true, false)`. A nicer way is above, with an `enum` - it'd be more readable to have `apply_block_messages(..., StoreEvents::Enabled)`. Additionally, it's more extendible.

What do you think?

Reply:
I agree. We can either use another `enum`. But could we make use of `bitflags`? (and shave off on param numbers).

Reply:
Actually, an enum is the best fit here.